### PR TITLE
VPC Block CSI Driver 5.1 release

### PR DESCRIPTION
**Pull request by GunaKKIBM**

### CODE REVIEW

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.1/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,723 @@
...
+                    path: vault-token
+                    expirationSeconds: 600 
+            - name: socket-dir
+              emptyDir: {}
+      volumeClaimTemplates: []
+  - apiVersion: snapshot.storage.k8s.io/v1
+    kind: VolumeSnapshotClass
+    metadata:
+      name: ibmc-vpcblock-snapshot
+      labels:
```
Comment (ambiknai) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.1/deployment.yaml:707]: We will make volumesnapshotclass "default"

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.1/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,723 @@
...
+      template:
+        metadata:
+          labels:
+            app: ibm-vpc-block-csi-driver
+          annotations:
+            autoUpdate: "AUTO-UPDATE-ENABLED"
+        spec:
+          serviceAccountName: ibm-vpc-block-node-sa
+          tolerations:
+            - operator: Exists
```
Comment (ambiknai) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.1/deployment.yaml:332]: We must add priorityClass here -system-node-critical

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.1/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+        matchLabels:
+          app: ibm-vpc-block-csi-driver
+      template:
+        metadata:
+          labels:
+            app: ibm-vpc-block-csi-driver
+          annotations:
+            autoUpdate: "AUTO-UPDATE-ENABLED"
+        spec:
+          serviceAccountName: ibm-vpc-block-controller-sa
```
Comment (ambiknai) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.1/deployment.yaml:532]: system-cluster-critical priorityClass must be added

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.1/custom-parameters.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    "mutable": "false"
+  }, {
+    "description": "The VPC regional endpoint of your VPC cluster in the format `https://region.iaas.cloud.ibm.com`. Example: `https://eu-de.iaas.cloud.ibm.com`. For more information, see https://ibm.biz/vpc-endpoints",
+    "displayname": "VPC IaaS endpoint",
+    "name": "g2_riaas_endpoint_url",
+    "required": "true",
+    "category": "config",
+    "mutable": "false"
+  }, {
+      "description": "The ID of the resource group where your VPC is located. To retrieve this value, run the 'ibmcloud is vpc VPC-ID' command and note the Resource group field.",
```
Comment (mssachan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.1/custom-parameters.json]: update description as `The ID of the resource group where your VPC is located. You can find your resource group by running the 'ibmcloud is vpc VPC-ID' command and note the Resource group field.` similar to 5.0 version

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.1/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,713 @@
...
+          annotations:
+            autoUpdate: "AUTO-UPDATE-ENABLED"
+        spec:
+          priorityClassName: system-node-critical
+          serviceAccountName: ibm-vpc-block-node-sa
+          tolerations:
+            - operator: Exists
+          initContainers:
+            - name: vpc-node-label-updater
+              image: icr.io/obs/storage/vpc-node-label-updater:v4.2.14
```
Comment (mssachan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.1/deployment.yaml]: use latest image with security fixes here: latest version as of today is v4.2.15
Use latest images at other places as well

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.0/metadata.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
   "name": "5.0",
   "revision": "14",
   "enabled": "true",
-  "status": "supported",
+  "status": "deprecated",
```
Comment (mssachan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.0/metadata.json]: Do not deprecate existing version until new version is already in PROD

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.1/metadata.json [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+      "name": "offering-name",
+      "value": "ibm-vpc-block-csi-driver"
+    },
+    {
+      "name": "offering-type",
+      "value": "non-sds"
+    },
+    {
+      "name": "provider-url",
+      "value": "IBM knowledge center: https://www.ibm.com/support/knowledgecenter/SSRQ8T_1.4.0/csi_block_storage_kc_welcome.html"
```
Comment (mssachan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.1/metadata.json]: This does not look to be correct link. Please refere version 5.0 for correct link.

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.1/metadata.json [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      "name": "support-msg",
+      "value": "For support contact IBM"
+    },
+    {
+      "name": "supported-envs",
+      "value": "IBM"
+    },
+    {
+      "name": "supported-ocp-versions",
+      "value": "4.10,4.11,4.12,4.13,4.14"
```
Comment (mssachan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.1/metadata.json]: put `space` after every comma.
`"4.10,4.11,4.12,4.13,4.14"` ----> `"4.10, 4.11, 4.12, 4.13, 4.14"`

File: config-templates/ibm/ibm-vpc-block-csi-driver/5.1/storage-class-parameters.json [hunk 9]
```diff
@@ -0,0 +1,155 @@
+[
+    {
```
Comment (mssachan) [inline config-templates/ibm/ibm-vpc-block-csi-driver/5.1/storage-class-parameters.json:2]: Please compare this file with 5.0 version file. Lot of regex being removed. Please review

**Review by mssachan — CHANGES_REQUESTED** [review]
Please compare all the files under new template version 5.1 with files of existing template version 5.0

**Comment by mssachan** [comment]
Also please update `changelog.json` with new version details in case you want to release this template in next stage push scheduled on 23rd November

**Review by mssachan — APPROVED** [review]
LGTM

**Pull request merged**